astropy/wcs/wcsapi/low_level_api.py

```diff
@@ -330,22 +356,35 @@ def __str__(self):
 
         # Find largest between header size and value length
         world_dim_width = max(9, len(str(self.world_n_dim)))
+        world_nam_width = max(9, max(len(x) if x is not None else 0 for x in self.world_axis_names))
         world_typ_width = max(13, max(len(x) if x is not None else 0 for x in self.world_axis_physical_types))
 
         s += (('{0:' + str(world_dim_width) + 's}').format('World Dim') + '  ' +
+              ('{0:' + str(world_nam_width) + 's}').format('Axis Name') + '  ' +
               ('{0:' + str(world_typ_width) + 's}').format('Physical Type') + '  ' +
                'Units\n')
 
         for iwrl in range(self.world_n_dim):
 
-            if self.world_axis_physical_types[iwrl] is not None:
-                s += (('{0:' + str(world_dim_width) + 'd}').format(iwrl) + '  ' +
-                      ('{0:' + str(world_typ_width) + 's}').format(self.world_axis_physical_types[iwrl]) + '  ' +
-                      '{:s}'.format(self.world_axis_units[iwrl] + '\n'))
+            if self.world_axis_names[iwrl] is None:
+                name = 'None'
+            else:
+                name = self.world_axis_names[iwrl]
+
+            if self.world_axis_physical_types[iwrl] is None:
+                typ = 'None'
             else:
-                s += (('{0:' + str(world_dim_width) + 'd}').format(iwrl) + '  ' +
-                      ('{0:' + str(world_typ_width) + 's}').format('None') + '  ' +
-                      '{:s}'.format('unknown' + '\n'))
+                typ = self.world_axis_physical_types[iwrl]
+
+            if self.world_axis_units[iwrl] is None:
+                unit = 'unknown'
+            else:
+                unit = self.world_axis_units[iwrl]
+
+            s += (('{0:' + str(world_dim_width) + 'd}').format(iwrl) + '  ' +
+                    ('{0:' + str(world_nam_width) + 's}').format(name) + '  ' +
+                    ('{0:' + str(world_typ_width) + 's}').format(typ) + '  ' +
+                    '{:s}'.format(unit + '\n'))
```

Comment:
I'd like to propose to NOT define `__str__` at the low level API and leave it to the concrete implementation. I think it would be better to define a new function for this purpose. As I mentioned in #9157, this overwrites the FITS WCS `__str__` and was a bit of a surprise. And it doesn't do anything for `gwcs` as `gwcs` defines its own `__str__`. If it's in a new function which doesn't overwrite something as basic as `__str__`, it will work for both packages using the API and can outline the structure in a common way.
Obviously it's not part of this PR, I mention it because this PR extends `__str__` and all involved are watching the issue.

Reply:
:+1: to implementing `__str__` as a function, I have uses for that as well.

I am in favour of changing the str for `WCS` but it probably needs more thought than how it ended up happening here. This should probably be a separate PR though.